src/integration/java/org/springframework/data/couchbase/CouchbaseWaitStrategy.java

```diff
@@ -0,0 +1,131 @@
+package org.springframework.data.couchbase;
+
+import static java.time.temporal.ChronoUnit.SECONDS;
+
+import java.time.Duration;
+import java.util.concurrent.Callable;
+import java.util.concurrent.TimeUnit;
+
+import org.rnorth.ducttape.ratelimits.RateLimiterBuilder;
+import org.rnorth.ducttape.unreliables.Unreliables;
+import org.testcontainers.containers.Container;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.wait.WaitStrategy;
+
+/**
+ * WaitStrategy for Couchbase containers which makes the Server node is
+ * initialized, RBAC user and default bucket is created.
+ */
+public class CouchbaseWaitStrategy implements WaitStrategy {
+
+	private Duration startupTimeout = Duration.of(60, SECONDS);
+	private Boolean rbacEnabled = false;
+
+	public CouchbaseWaitStrategy(String serverVerion) {
+		String[] verionSplit = serverVerion.split("\\.");
+		int major = Integer.parseInt(verionSplit[0]);
+		if (major >= 5) {
+			rbacEnabled = true;
+		}
+	}
+
+	private void checkResult(Container.ExecResult result, String command) throws Exception {
```

Comment:
do we know with couchbase-cli that once it returns its done?

Reply:
The cluster init and bucket create is synchronous. The user create is async, but atleast it is created at the ns_server end. There is a rest retryer to make sure the query service is up. So we know that the cluster is initialized and mds is configured.